src/groupeddataframe/utils.jl

```diff
@@ -303,6 +306,21 @@ function row_group_slots(cols::NTuple{N, AbstractVector},
     else
         sorted = false
     end
+
+    lg = length(groups)
+    nt = Threads.nthreads()
+    use_threading = nt > 1 && lg > 1_000_000 && ngroups < lg * (0.5 - 1 / (2 * nt)) / (2 * nt)
```

Comment:
```suggestion
    # TODO: in the future when `ngroups` is large relative to `lg` improve the algorithm to
    # 1. increase chunk size so that we do not use all threads
    # 2. initialize `seen` elements lazily (only when actually the task on a particular thread is spawned)
    # 3. in reduction process only these `seen` elements that were initialized
    # (finding optimal rule for this is hard so it is left for later)
    use_threading = nt > 1 && lg > 1_000_000 && ngroups < lg * (0.5 - 1 / (2 * nt)) / (2 * nt)
```